### PR TITLE
Add MonadError instance for TemplatesT

### PR DESCRIPTION
**Pull request by tombusby**

### CODE REVIEW

File: src/Text/StringTemplates/Templates.hs [hunk 1]
```diff
@@ -170,3 +179,11 @@ instance (Applicative m, Monad m) => TemplatesMonad (TemplatesT m) where
...
     (_, ts) <- ask
     return $ localizedVersion lang ts
+
+instance MonadError e m => MonadError e (TemplatesT m) where
+  throwError = lift . throwError
+  catchError = liftCatch catchError
+    where
+      liftCatch :: Catch e m a -> Catch e (TemplatesT m) a
+      liftCatch f m h = TemplatesT . ReaderT $ \r ->
+        f (runReaderT (unTT m) r) (\e -> runReaderT (unTT $ h e) r)
```
Comment (arybczak) [inline src/Text/StringTemplates/Templates.hs]: You don't need to strip the ReaderT here

**Comment by arybczak** [comment]
Thanks. I simplified the instance and added changelog, will make a release in a while.

**Comment by arybczak** [comment]
https://hackage.haskell.org/package/kontrakcja-templates-0.12

File: src/Text/StringTemplates/Templates.hs (hunk 1 above)
Comment (tombusby) [reply src/Text/StringTemplates/Templates.hs]: Oh yeah, of course, cool :)

**Pull request merged**